com.unity.render-pipelines.core/Editor/Volume/Drawers/TextureParameterDrawer.cs

```diff
@@ -0,0 +1,132 @@
+using System;
+using System.Reflection;
+using UnityEngine;
+using UnityEngine.Rendering;
+using Object = UnityEngine.Object;
+
+namespace UnityEditor.Rendering
+{
+    internal static class TextureParameterHelper
+    {
+        static readonly Type k_ObjectFieldValidator = Type.GetType("UnityEditor.EditorGUI+ObjectFieldValidator,UnityEditor");
+        static readonly MethodInfo k_ValidateObjectFieldAssignment = Type.GetType("UnityEditor.EditorGUI,UnityEditor").GetMethod("ValidateObjectFieldAssignment", BindingFlags.Static | BindingFlags.NonPublic);
+        static readonly MethodInfo k_EditorGUI_DoObjectField = Type.GetType("UnityEditor.EditorGUI,UnityEditor")
+            .GetMethod("DoObjectField", BindingFlags.Static | BindingFlags.NonPublic, null,
+                    new Type[] { typeof(Rect), typeof(Rect), typeof(int), typeof(Object), typeof(Object), typeof(Type), typeof(Type), typeof(SerializedProperty), k_ObjectFieldValidator, typeof(bool), typeof(GUIStyle) }, null);
+
+        internal delegate Object ObjectFieldValidator(Object[] references, Type objType, SerializedProperty property, int options);
+
+        internal static Object ValidateObjectFieldAssignment(Object[] references, Type objType, SerializedProperty property, int options)
+        {
+            return (Object)k_ValidateObjectFieldAssignment.Invoke(null, new object[] { references, objType, property, options });
+        }
+
+        internal static Delegate CastValidator(ObjectFieldValidator validator)
+        {
+            return DelegateUtility.Cast(validator, k_ObjectFieldValidator);
+        }
+
+        internal static void DoObjectField(SerializedProperty property, GUIContent label, Type type1, Type type2, Delegate validator)
+        {
+            Rect r = EditorGUILayout.GetControlRect(false, EditorGUIUtility.singleLineHeight);
+            DoObjectField(r, property, label, type1, type2, validator);
+        }
+
+        internal static void DoObjectField(Rect r, SerializedProperty property, GUIContent label, Type type1, Type type2, Delegate validator)
+        {
+            int id = GUIUtility.GetControlID(FocusType.Keyboard, r);
+            r = EditorGUI.PrefixLabel(r, id, label);
+
+            var parameters = new object[] { r, r, id, null, null, type1, type2, property, validator, false, EditorStyles.objectField };
+            k_EditorGUI_DoObjectField.Invoke(null, parameters);
```

Comment:
[Performence] You should compile once using linq tree at static construction instead of researching the method on the object each time with Invoke. Here it is a static one. Not 100% sure there is perf to gain beside the GC of the object array.

Reply:
for this one I didn't manage to do it because the function DoObjectField takes an argument internal to the engine
So I can't define a delegate at compile time for this function signature (or I don't know how to)